src/Modal.js

```diff
@@ -277,6 +277,11 @@ const Modal = React.createClass({
     }
   },
 
+  componentWillUnmount() {
+    this.onHide();
+    this.props.onHide();
```

Comment:
oh right, that's my fault indeed.

I cannot think of something better than creating a dedicated method with a flag that would invoke `this.props.onHide()` once.

But I'm not sure if there are scenarios when the component can be shown twice without recreating from the scratch.